### ISSUE TITLE
Kernel restarts after memory error, but no error message displays.

### ISSUE THREAD

**Issue by klazaj** [jupyterlab/jupyterlab#9200]
I moved from Jupyter Notebook to Jupyter Lab, and now when the Kernel dies due to memory allocation error, there's no error message saying 'Kernel died'. I only see that the number disappears from the running cell:

![image](https://user-images.githubusercontent.com/22456202/96860891-a192c380-1463-11eb-8e89-b4cfb259d308.png)

The message I get in the console is:

`KernelRestarter: restarting kernel (1/5), keep random ports kernel 7bb53f39-b0a8-4e32-bbe6-e6dbe7c9c344 restarted`

Why is this happening? How can I display the error on Jupyter Lab too?

I used to get this message shown in Jupyter Notebook:

![image](https://user-images.githubusercontent.com/22456202/96862294-917be380-1465-11eb-99b9-4fb0c61bc7fb.png)

More info: 
- Operating System and version: Ubuntu 20.04
- Browser and version: Google Chrome 86.0.4240.111
- JupyterLab version: 2.2.8

**Comment by jasongrout** [comment]
Can you fill out the information from the new issue template, like JupyterLab version, etc.?

**Comment by jasongrout** [comment]
(I'm pretty sure this is fixed in recent jlab versions.)

**Comment by klazaj** [comment]
@jasongrout I added OS, browser, and Lab version.

**Comment by dclong** [comment]
I'm encountering the same issue in JupyterLab 2.2.9

**Comment by jasongrout** [comment]
I'm able to reproduce this in 3.0.5 by running this, for example:

```python
import numpy as np
arr = []
for x in range(10000):
    my_array = np.zeros(2**30)
    my_array += 10
    arr.append(my_array)
```

The kernel is restarted, and sends a restarting message to the frontend, but a dialog never pops up saying the kernel was restarted.

It seems that #8562 broke this, and we did not have tests to catch the breakage in the review of that PR. In particular, https://github.com/jupyterlab/jupyterlab/pull/8562/files#diff-1acd3b371270660fed12e4595862cfb74ab7e973dd853b9caac92dcb34b6ca99R1314 changed JupyterLab to listen to a non-existent 'autorestarting' message instead of the correct 'restarting' message from the server.

CC @echarles

**Comment by echarles** [comment]
@jasongrout Do you think reverting `if (executionState === 'autorestarting')` to `if (executionState === 'restarting')`  would solve the issue?

**Comment by jasongrout** [comment]
That was my first thought - but notice that #8562 introduced a new block of almost identical code handling the 'restarting' case just below, with one small change. It made me think that we need to think carefully through the ramifications because the 'restarting' status took on slightly different meaning with different effects after #8562. Anyways, I just wasn't sure what the ramifications of just changing that if condition.
